launcher/src/test/java/org/wildfly/core/launcher/CommandBuilderTest.java

```diff
@@ -90,8 +88,13 @@ public void testStandaloneBuilder() {
 
         Assert.assertTrue("Missing -secmgr option", commands.contains("-secmgr"));
 
-        Assert.assertTrue("Missing --add-exports option", commands.contains("--add-exports=jdk.unsupported/sun.reflect=ALL-UNNAMED"));
-
+        // If we're using Java 9+ ensure the modular JDK options were added
+        if (Environment.isModularJavaHome(JAVA_HOME)) {
+            assertArgumentExists(commands, "--add-exports=java.base/sun.nio.ch=ALL-UNNAMED");
+            assertArgumentExists(commands, "--add-exports=jdk.unsupported/sun.reflect=ALL-UNNAMED");
+            assertArgumentExists(commands, "--add-exports=jdk.unsupported/sun.misc=ALL-UNNAMED");
+            assertArgumentExists(commands, "--add-modules=java.se");
+        }
```

Comment:
We can also add `else` block here and assert that arguments doesn't exist in non-modular JDK, right?

Reply:
Yeah that's probably a good idea. I'll update the PR, thanks!